pkg/sql/sem/tree/function_name.go

```diff
@@ -63,6 +63,9 @@ type FunctionReferenceResolver interface {
 // ResolvableFunctionReference implements the editable reference call of a
 // FuncExpr.
 type ResolvableFunctionReference struct {
+	// CallsiteName is the original UnresolvedName used at the function callsite.
+	// It may be unset if the function is unresolved.
+	CallsiteName *UnresolvedName
```

Comment:
Something about this field name bothers me. I think your approach here is sound. Also, I wonder if we should make this field unexported and have the act of resolving which rewrites this in place to do the swapping out into the field. Then have a method which uses the unexported field if it's populated and otherwise delegates to the underlying function reference.